src/abi.h

```diff
@@ -32,6 +32,14 @@ typedef enum {
     NETPLAN_OPTIONAL_STATIC  = 1<<4,
 } NetplanOptionalAddressFlag;
 
+typedef enum {
+    NETPLAN_CRITICAL_FALSE,
+    NETPLAN_CRITICAL_TRUE,
+    NETPLAN_CRITICAL_STATIC,
+    NETPLAN_CRITICAL_DHCP,
+    NETPLAN_CRITICAL_DHCP_ON_STOP,
+} NetplanCriticalOption;
```

Comment:
nitpick: we might just want to call that `NetplanCritical` to stay more in line with the other enums. Everything is an "Option", so that suffix feels kind of redundant.

PS: please also add and use a `NETPLAN_CRITICAL_UNSET` option, as suggested by @daniloegea 

Note: changing the `gboolean` type to an enum should not break our ABI, as both will be mapped to a `gint` in the end.